File: 206. Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode output = null;
        
        if (head == null) {
            return head;
        } else {
            while (head != null) {
                ListNode nxt = head.next; // save next 
                head.next = output; // next is now the previous
                output = head; // moved previous into current
                head = nxt; // move head onto next
            }
        }
        return output;
    }
}

File: 206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        output = None
        
        if not head:
            return head
        
        while head:
            tmp = head.next
            head.next = output
            output = head
            head = tmp
        
        return output
